File: solutions/beecrowd/2221/2221.cpp
#include <cstdint>
#include <iostream>

std::int16_t calculate_value(
    std::int16_t attack,
    std::int16_t defense,
    std::int16_t bonus,
    std::int16_t level) {
    std::int16_t result = (attack + defense) / 2;
    return result + (level % 2 == 0 ? bonus : 0);
}

int main() {
    std::int16_t t, a, d, l, bonus, dabriels_value, guartes_value;

    std::cin >> t;

    while (t--) {
        std::cin >> bonus;
        std::cin >> a >> d >> l;
        dabriels_value = calculate_value(a, d, bonus, l);
        std::cin >> a >> d >> l;
        guartes_value = calculate_value(a, d, bonus, l);

        if (dabriels_value < guartes_value) {
            std::cout << "Guarte" << std::endl;
        } else if (guartes_value < dabriels_value) {
            std::cout << "Dabriel" << std::endl;
        } else {
            std::cout << "Empate" << std::endl;
        }
    }

    return 0;
}

File: solutions/beecrowd/2221/2221.py
import sys

input = sys.stdin.readline
output = sys.stdout.write


def calculate_value(attack, defense, bonus, level):
    result = (attack + defense) / 2
    return result + (bonus if level % 2 == 0 else 0)


t = int(input())
for _ in range(t):
    bonus = int(input())
    attack, defense, level = map(int, input().split())
    dabriels_value = calculate_value(attack, defense, bonus, level)
    attack, defense, level = map(int, input().split())
    guartes_value = calculate_value(attack, defense, bonus, level)

    if guartes_value > dabriels_value:
        output('Guarte\n')
    elif dabriels_value > guartes_value:
        output('Dabriel\n')
    else:
        output('Empate\n')
